OpenNefia.Content/CharaMake/CharaMakeLogic.cs

```diff
@@ -48,12 +57,24 @@ public void RunCreateChara()
                     Logger.DebugS("charamake", $"Character creation complete, values:" + Environment.NewLine 
                         + string.Join(Environment.NewLine, data.CharaData.SelectMany(x => x.Value.Select(y => $"{y.Key}: {y.Value}"))));
 
+                    var globalMap = _mapManager.CreateMap(1, 1, MapId.Global);
+                    var globalMapSpatial = _entityManager.GetComponent<SpatialComponent>(globalMap.MapEntityUid);
+
+
+                    var playerEntity = _entityManager.CreateEntityUninitialized(Protos.Chara.Player);
+                    var areaSpatial = _entityManager.GetComponent<SpatialComponent>(playerEntity);
+                    areaSpatial.AttachParent(globalMapSpatial);
+
+                    var entityGen = EntitySystem.Get<IEntityGen>();
+                    _entityManager.InitializeComponents(playerEntity);
+                    _entityManager.StartComponents(playerEntity);
                     foreach(var creationStep in steps)
                     {
-                        //TODO actually apply changes
-                        creationStep.ApplyStep(/* character entity (?) */);
+                        creationStep.ApplyStep(playerEntity);
                         creationStep.Dispose();
```

Comment:
I think the player should be able to back out of this and return to the last step, so the steps shouldn't be disposed here.

Reply:
changed in 2e3ad22